src/bc_protocol.rs

```diff
@@ -245,7 +245,7 @@ impl BcCamera {
 
     pub fn start_video(
         &self,
-        data_out: &mut dyn Write,
```

Comment:
Just caught this re: your comment. Yes, please keep as `dyn Write` (and add a second argument for the audio stream) :+1:

Reply:
I have to also pass up that the format changes that set gst to use the right launch str. maybe via a hints callback. Would that be acceptable?

Reply:
Oh, good point. In that case I think the callback would probably be uglier; you can leave this. In general I would love for this to be a generic library on crates.io but that refactor is _definitely_ down the road.